tests/python/pants_test/backend/codegen/tasks/test_simple_codegen_task.py

```diff
@@ -0,0 +1,230 @@
+# coding=utf-8
+# Copyright 2015 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+from textwrap import dedent
+
+from pants.backend.codegen.register import build_file_aliases as register_codegen
+from pants.backend.codegen.tasks.simple_codegen_task import SimpleCodegenTask
+from pants.backend.core.register import build_file_aliases as register_core
+from pants.backend.jvm.targets.java_library import JavaLibrary
+from pants.backend.jvm.targets.jvm_target import JvmTarget
+from pants.base.build_file_aliases import BuildFileAliases
+from pants.base.payload import Payload
+from pants_test.tasks.task_test_base import TaskTestBase
+
+
+class SimpleCodegenTaskTest(TaskTestBase):
+  @classmethod
+  def task_type(cls):
+    return SimpleCodegenTask
+
+  @property
+  def alias_groups(self):
+      return register_core().merge(register_codegen()).merge(BuildFileAliases.create({
+        'dummy_library': SimpleCodegenTaskTest.DummyLibrary
+      }))
+
+  def _create_subclass_task(self, subclass_type, target_roots=None, **options):
+    scope = 'test_scope_{}'.format(type(subclass_type))
+    task_type = self.synthesize_task_subtype(subclass_type, scope)
+    self.set_options_for_scope(scope, **options)
+    context = self.context(for_task_types=[task_type], target_roots=target_roots)
+    return task_type(context, self._test_workdir)
+
+  def _create_dummy_task(self, target_roots, **options):
+    SimpleCodegenTaskTest.DummyGen.TEST_BASE = self
```

Comment:
There are 2 style issues I have with this:

1) Names in caps are generally constants
2) From reading DummyGen, its not clear to me what the name `TEST_BASE` means.
3) Its also not obvious how TEST_BASE gets set.
4) Its a class variable but potentially gets reset to different instances of TestCase.   It should be a part of the instance.

I suggest you use a method to set these values in the class.    In this case, it looks like you are passing in `self` to get the methods on `unittest.TestCase`  Maybe make the member name reflect the class name you want to invoke methods on.

task = create_task(self.context())
task.set_test_case(self)

...

class DummyGen(...):
  ...

  def set_test_case(self, test_case):
    self._test_case = test_case

   ... 

   self._test_case.assertEquals(...)